libwallet/assets_manager.go

```diff
@@ -909,19 +911,27 @@ func (mgr *AssetsManager) DexcReady() bool {
 // InitializeDEX initializes mgr.dexc.
 func (mgr *AssetsManager) InitializeDEX(ctx context.Context) {
 	logDir := filepath.Dir(mgr.LogFile())
-	dexc, err := dexc.Start(ctx, mgr.RootDir(), mgr.GetLanguagePreference(), logDir, mgr.GetLogLevels(), mgr.NetType(), 0 /* TODO: Make configurable */)
+	dexcl, err := dexc.Start(ctx, mgr.RootDir(), mgr.GetLanguagePreference(), logDir, mgr.GetLogLevels(), mgr.NetType(), 0 /* TODO: Make configurable */)
 	if err != nil {
 		log.Errorf("Error starting dex client: %v", err)
 		return
 	}
 
 	mgr.dexcMtx.Lock()
-	mgr.dexc = dexc
+	mgr.dexc = dexcl
 	mgr.dexcMtx.Unlock()
 
-	err = mgr.PrepareDexSupportForDcrWallet()
+	// Prepare support for DCR wallet.
+	err = mgr.PrepareDexSupportForDCRWallet()
```

Comment:
Can this mgr.InitializeDEX method be called multiple times while the app is open? If not, then `mgr.PrepareDexSupportForDCRWallet()` should not need to worry about duplicate registrations. Also, I don't see why we would want to initialize dexc multiple times.

Reply:
> Can this mgr.InitializeDEX method be called multiple times while the app is open?

Yes. We can't control what happens in dex core so we try to restart the dex client if it's not yet started.